src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -280,6 +281,45 @@ static boolean avoidWalkingInto(Block block) {
                 || block == Blocks.WEB;
     }
 
+    static boolean isOverMagma(IPlayerContext ctx, BetterBlockPos src, BetterBlockPos dest) {
+        int x = ctx.playerFeet().x;
+        int y = ctx.playerFeet().y;
+        int z = ctx.playerFeet().z;
+        BlockStateInterface bsi = new BlockStateInterface(ctx);
+
+        if (hasFrostWalker(ctx)) { //if we are wearing frostwalker boots magma can go fuck itself
+            return false;
+        }
+        if (isOverMagma(bsi, x, y, z) || isOverMagma(bsi, dest.x, y, dest.z)) { //we are or going to stand on magma
+            return true;
+        }
+        if (isOverMagma(bsi, dest.x, dest.y, dest.z) && y < src.y) {            //we drop down on a magma block
+            return true;
+        }
+        if ((src.x != dest.x && src.z != dest.z)) {                             //we move diagonaly
+            return isOverMagma(bsi, src.x, y, dest.z) || isOverMagma(bsi, dest.x, y, dest.z) || isOverMagma(bsi, dest.x, y, src.z); //over a magmablock
+        }
+        return false; //if we get this far there is no magma
+    }
+    static boolean isOverMagma(BlockStateInterface bsi, int x, int y, int z) {
+        return bsi.get0(x,y-1,z).getBlock() == Blocks.MAGMA;
+    }
+
+    static boolean hasFrostWalker(CalculationContext context) {
+        return hasFrostWalker(context.getBaritone().getPlayerContext());
+    }
+
+    static boolean hasFrostWalker(IPlayerContext ctx) {
+        boolean frostwaker = false;
+        NBTTagList nbtTagList = ctx.player().inventory.player.inventory.player.inventory.player.inventory.player.inventory.armorInventory.get(0).getEnchantmentTagList(); //get all enchantments of the players boots
+        for (int i = 0; i<nbtTagList.tagCount();i++) {
+            if(nbtTagList.get(i).toString().contains("id:9s")) { //this is very hacky pls advice on how to test for frostwalker enchantment the proper way
+                frostwaker = true;
```

Comment:
this could probably be made into a early return statment but since boots can hold at moste 9 enchantments includiung curses im not sure if its worth it.

Reply:
Is this what you are looking for?
https://github.com/ZacSharp/baritone/blob/fb814e912dadf3f6e34358354a86b54674103081/src/main/java/baritone/pathing/movement/CalculationContext.java#L104
or maybe `net.minecraft.enchantment.EnchantmentHelper.hasFrostWalkerEnchantment`?